### PR TITLE
Make nft.burns trino compatible + clean up comments and unnecessary casts

### PR DESCRIPTION
**Pull request by belen-pruvost**
- Making nft.burns trino compatible by explicitly casting token_id in x2y2_ethereum.events as string
- Removing explicit cast on not base level view
- Removing left over comments from https://github.com/duneanalytics/spellbook/pull/2182

### CODE REVIEW

**Comment by belen-pruvost** [comment]
As I open this PR, `dbt run --select x2y2_ethereum_events` is still running locally for me

**Comment by belen-pruvost** [comment]
Locally I've run successfully:
- dbt run --select x2y2_ethereum_events   
- dbt run --select x2y2_ethereum_burns

However `dbt run --select nft_burns` is still pending.

**Comment by 0xRobin** [comment]
> * Making nft.burns trino compatible by explicitly casting token_id in x2y2_ethereum.events as string

There's an open PR (https://github.com/duneanalytics/spellbook/pull/2118) that fully rewrites the x2y2 events.
Here's the line that will need an explicit cast wrapper to string (depending on the order these PRs are merged).
https://github.com/duneanalytics/spellbook/blob/7eca58d0b0efe1e225a706b4f88c0a5cfbac3887/models/x2y2/ethereum/x2y2_ethereum_events.sql#L109

> I believe the failing pre commit hook check error is unrelated to my changes: Missing model (ref) model

That's correct, for this explicit test it's resolved in this open PR (https://github.com/duneanalytics/spellbook/pull/2172)

**Comment by belen-pruvost** [comment]
`dbt run --select nft_burns` has finished running successfully, so I guess this PR should hopefully be good for review

File: models/nft/nft_trades.sql [hunk 1]
```diff
@@ -35,21 +35,21 @@ FROM (
         amount_usd,
         token_standard,
         trade_type,
-        number_of_items, --team-query cast as decimal: CAST(number_of_items AS DECIMAL(38,0)) AS number_of_items,
```
Comment (jeff-dude) [inline models/nft/nft_trades.sql:38]: thank you 🥲

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**